includes/Core/Email_Reporting/Email_Report_Data_Section_Part.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Email_Reporting\Email_Report_Data_Section_Part
+ *
+ * @package   Google\Site_Kit\Core\Email_Reporting
+ * @copyright 2025 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Email_Reporting;
+
+use InvalidArgumentException;
+
+/**
+ * Single email report section part.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Email_Report_Data_Section_Part {
+
+	/**
+	 * Unique section key.
+	 *
+	 * @var string
+	 */
+	private $section_key;
+
+	/**
+	 * Section title.
+	 *
+	 * @var string
+	 */
+	private $title;
+
+	/**
+	 * Labels for the section rows/series.
+	 *
+	 * @var array
+	 */
+	private $labels;
+
+	/**
+	 * Values formatted as strings for output.
+	 *
+	 * @var array
+	 */
+	private $values;
+
+	/**
+	 * Optional trends matching values.
+	 *
+	 * @var array|null
+	 */
+	private $trends;
+
+	/**
+	 * Optional date range data.
+	 *
+	 * @var array|null
+	 */
+	private $date_range;
+
+	/**
+	 * Optional dashboard deeplink URL.
+	 *
+	 * @var string|null
+	 */
+	private $dashboard_link;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string      $section_key    Unique section key.
+	 * @param string      $title          Section title.
+	 * @param array       $labels         Labels (strings) for the section rows/series.
+	 * @param array       $values         Values (already formatted to strings for output).
+	 * @param array|null  $trends         Optional trends (strings) matching values.
+	 * @param array|null  $date_range     Optional date range array with 'startDate' and 'endDate'.
+	 * @param string|null $dashboard_link Optional dashboard deeplink.
+	 *
+	 * @throws InvalidArgumentException When validation fails.
+	 */
+	public function __construct(
+		$section_key,
+		$title,
+		array $labels,
+		array $values,
+		$trends = null,
+		$date_range = null,
+		$dashboard_link = null
+	) {
+		$section_key = is_string( $section_key ) ? $section_key : '';
+		$title       = is_string( $title ) ? $title : '';
+
+		if ( '' === $section_key ) {
+			throw new InvalidArgumentException( 'section_key must be a non-empty string' );
+		}
+		if ( '' === $title ) {
+			throw new InvalidArgumentException( 'title must be a non-empty string' );
+		}
+
+		$this->labels = array_map( 'strval', $labels );
+		$this->values = array_map( 'strval', $values );
+
+		if ( ! is_array( $labels ) ) {
+			throw new InvalidArgumentException( 'labels must be an array' );
+		}
+		if ( ! is_array( $values ) ) {
+			throw new InvalidArgumentException( 'values must be an array' );
+		}
```

Comment:
This seems to be logically inverted, mapping is done before knowing the input is valid. `array_map()` will issue a warning or fatal error if `$labels` or `$values` are not arrays, so the validation after the map will never be reached safely.

```suggestion
		if ( ! is_array( $labels ) ) {
			throw new InvalidArgumentException( 'labels must be an array' );
		}
		if ( ! is_array( $values ) ) {
			throw new InvalidArgumentException( 'values must be an array' );
		}
		
		$this->labels = array_map( 'strval', $labels );
		$this->values = array_map( 'strval', $values );
```